testing/tests/DevExpress.ui.widgets.dataGrid/columnChooserModule.tests.js

```diff
@@ -1156,5 +1156,58 @@ QUnit.module('Column chooser', {
         assert.strictEqual(popupInstance.option('width'), 500, 'changed width');
         assert.strictEqual(popupInstance.option('height'), 600, 'changed height');
     });
+
+    // T907091
+    [true, false].forEach(useBeginEndUpdate => {
+        QUnit.test(`ColumnChooser's item captions should be updated if column captions were changed (${useBeginEndUpdate ? 'with' : 'without'} optimization)`, function(assert) {
+            // arrange
+            const $testElement = $('#container');
+
+            this.options.columnChooser.mode = 'select';
+            $.extend(this.columns, [{ caption: 'Column 1', index: 0, visible: true, showInColumnChooser: true }, { caption: 'Column 2', index: 1, visible: true, showInColumnChooser: true }]);
+            this.setTestElement($testElement);
+
+            sinon.spy(this.columnChooserView, '_renderTreeView');
+
+            // act
+            this.columnChooserView.showColumnChooser();
+            this.clock.tick(1000);
+
+            assert.strictEqual(this.columnChooserView._renderTreeView.callCount, 1, 'treeview is rendered');
+
+            const optionNames = {
+                caption: true,
+                length: 1
+            };
+
+            if(useBeginEndUpdate) {
+                for(let i = 0; i < 2; i++) {
+                    this.beginUpdate();
+                    this.columnsController.columnOption(i, 'caption', 'new caption');
+                    this.endUpdate();
+                }
+                this.columnsController.columnsChanged.fire({
+                    optionNames
+                });
+            } else {
```

Comment:
It is not quite clear why **beginUpdate** and **endUpdate** methods should be used here. I think that the test result will be the same if you remove them.